include/deal.II/base/symmetric_tensor.h

```diff
@@ -2096,6 +2099,42 @@ SymmetricTensor<rank,dim,Number>::serialize(Archive &ar, const unsigned int)
 
 
 /**
+ * Add two symmetric tensors. If possible, you should use <tt>operator +=</tt>
+ * instead since this does not need the creation of a temporary.
+ *
+ * @relates SymmetricTensor
+ */
+template <int rank, int dim, typename Number, typename OtherNumber>
+inline
+SymmetricTensor<rank, dim, typename ProductType<Number, OtherNumber>::type>
+operator + (const SymmetricTensor<rank, dim, Number>      &left,
```

Comment:
I will open up a github issue. As I said, I __do__ want to come back to this but I don't think that its the best use of my time right now. I need more time to familiarise myself with how the Sacado AD classes and expression templates work. I'd rather get the auto-differentiation helper classes into the library first.

I'll dump the core part of my prototype code into the forthcoming issue. I did use `decltype()` etc. but as I said, having something simple like (off the top of my head)
```c++
template<typename Number1, typename Number2>
struct AdditionType
{
  typedef decltype(std::declval<Number1>() + std::declval<Number2>()) type;
};

template<typename Number1, typename Number2>
typename AdditionType<Number1,Number2>::type
add_1 (const Number1 &a, const Number2 &b)
{
   typename AdditionType<Number1,Number2>::type out;
  out = internal::NumberType<Number>::value(0.0);
   out += a;
   out += b;
   return out;
}

template<typename Number1, typename Number2>
typename AdditionType<Number1,Number2>::type
add_2 (const Number1 &a, const Number2 &b)
{
   typename AdditionType<Number1,Number2>::type out (a);
   out += b;
   return out;
}
```
does not necessarily work for all combinations of Sacado related `Number1`,`Number2` and `AdditionType::type`.